hw-36.ts

```diff
@@ -0,0 +1,198 @@
+        boxOffice: "$302,305,431",
+        production: "Heyday Films, Warner Bros.",
+    },
+    {
+        title: "Harry Potter and the Sorcerer's Stone",
+        year: 2001,
+        released: "16 Nov 2001",
+        runtime: "152 min",
+        genre: ["Adventure", "Family", "Fantasy"],
+        director: "Chris Columbus",
+        writer: "J.K. Rowling, Steve Kloves",
+        actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+        plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+        country: "United Kingdom, United States",
+        poster: "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+        imdbRating: 7.6,
+        imdbVotes: 684604,
+        boxOffice: "$318,087,620",
+        production: "1492 Pictures, Heyday Films, Warner Brothers",
+    }
+
+]
+
+// 1. Собрать в массив все жанры фильмов ( без повторений)
+
+console.log("1.\n")
+const getGenre = (array: Film[]) => {
+    const newArr = array.map(film => film.genre)
+    const genresArr = newArr.reduce((acc, genres) => [...acc, ...genres], [])
+    const result = genresArr.filter((genre, index, genres) => genres.indexOf(genre) === index)
+    return result
+}
+
+console.log(getGenre(films))
+
+// 2. Собрать в массив всех актеров всех фильмов (без повторений)
+console.log("\n2.\n")
+
+const getActors = (array: Film[]) => {
+    const newArr = array.map(film => film.actors)
+    const actorsArr = newArr.reduce((acc, actors) => [...acc, ...actors], [])
+    const result = actorsArr.filter((actor, index, actors) => actors.indexOf(actor) === index)
+    return result
+}
+
+console.log(getActors(films))
+
+// 3. 
+console.log('\n3.\n')
+
+const getSortedRaiting = (array: Film[]) => {
+    return array.sort((a, b) => b.imdbRating - a.imdbRating)
+}
+
+console.log(getSortedRaiting(films))
+
+// 4. 
+console.log('\n4.\n')
+
+const getNewFilmsArray = (array: Film[]) => {
+    return array.map(film => ({ id: film.id, title: film.title, released: film.released, plot: film.plot }))
+}
+
+console.log(getNewFilmsArray(films))
+
+// 5. 
+console.log('\n5.\n')
+
+const getFilmYear = (array: Film[], year: number) => {
+    return array.filter((film) => film.year === year)
+}
+
+console.log(getFilmYear(films, 2011))
+
+// 6. 
+console.log('\n6.\n')
+
+const hasStrNameFilm = (array: Film[], str: string) => {
+    return array.filter((film) => film.title.includes(str))
+}
+
+console.log(hasStrNameFilm(films, 'Potter'))
+
+// 7. 
+console.log('\n7.\n')
+
+const hasNameFilm = (array: Film[], str: string) => {
+    return array.filter((film) => film.title.includes(str) || film.plot.includes(str))
+}
+
+console.log(hasNameFilm(films, 'Star'))
+
+// 8. 
+console.log('\n8.\n')
+
+function filterFilmsByProperty(films: Film[], property: keyof Film, value: string | number) {
```

Comment:
thanks for `property: keyof Film`!
you only one who remembers it. Including me)